### ISSUE TITLE
【consult🙋‍♂️】 some confusion about CustomVisionTransformer.forward_features

### ISSUE THREAD

**Issue by TITC** [lukas-blecher/LaTeX-OCR#129]
Thanks for your response these days and I learned a lot from your repo.

When I read code in [this part](https://github.com/lukas-blecher/LaTeX-OCR/blob/e02c395140f76fe7f0b23314ba0caa63d02a0e88/pix2tex/models.py#L69-L76), I can understand what the operation has done at the API level, but I don't know why should do this thing and the theory behind it.

1. the reason of concatenate [`cls_tokens` and `x`](https://github.com/lukas-blecher/LaTeX-OCR/blob/e02c395140f76fe7f0b23314ba0caa63d02a0e88/pix2tex/models.py#L73-L74)
```python
x = torch.cat((cls_tokens, x), dim=1)
```
2.  the special purpose of `self.width//self.patch_size-w`, can I replace it with a random number, such as 2, or 3?
```python
pos_emb_ind = repeat(torch.arange(h)*(self.width//self.patch_size-w), 'h -> (h w)', w=w)+torch.arange(h*w)
```

I know `torch.arange(h)` generate a list from `0` to `h-1`, and then use `(self.width//self.patch_size-w` the upscale the gap between 2 rows. Then use  repeat to transfer (h,1) dim tensor to (h,w) tensor and squeeze it into a vector. These operations differ row's position information in the image, last plus a list `torch.arange(h*w)` from `0`-`h*w-1`, differ element at each row.

example 
- h=12
- self.width//self.patch_size=10
- self.width//self.patch_size-w =3
- w=10

assume each column chunk separated by blank cell, 
- first chunk denotes `torch.arange(h)`
- second chunk represents `torch.arange(h)*(self.width//self.patch_size-w)`
- third chunk denotes `repeat(torch.arange(h)*(self.width//self.patch_size-w), 'h -> h w', w=w)`

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Yuhang/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Yuhang/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

0 |   |   | 0 |   |   | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 |   |   | 3 |   |   | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3 | 3
2 |   |   | 6 |   |   | 6 | 6 | 6 | 6 | 6 | 6 | 6 | 6 | 6 | 6
3 |   |   | 9 |   |   | 9 | 9 | 9 | 9 | 9 | 9 | 9 | 9 | 9 | 9
4 |   |   | 12 |   |   | 12 | 12 | 12 | 12 | 12 | 12 | 12 | 12 | 12 | 12
5 |   |   | 15 |   |   | 15 | 15 | 15 | 15 | 15 | 15 | 15 | 15 | 15 | 15
6 |   |   | 18 |   |   | 18 | 18 | 18 | 18 | 18 | 18 | 18 | 18 | 18 | 18
7 |   |   | 21 |   |   | 21 | 21 | 21 | 21 | 21 | 21 | 21 | 21 | 21 | 21
8 |   |   | 24 |   |   | 24 | 24 | 24 | 24 | 24 | 24 | 24 | 24 | 24 | 24
9 |   |   | 27 |   |   | 27 | 27 | 27 | 27 | 27 | 27 | 27 | 27 | 27 | 27
10 |   |   | 30 |   |   | 30 | 30 | 30 | 30 | 30 | 30 | 30 | 30 | 30 | 30
11 |   |   | 33 |   |   | 33 | 33 | 33 | 33 | 33 | 33 | 33 | 33 | 33 | 33

</body>

</html>
